xwiki-commons-core/xwiki-commons-velocity/src/main/java/org/xwiki/velocity/XWikiVelocityContext.java

```diff
@@ -110,6 +113,17 @@ private void warnDeprecatedBinding(String binding, ForeachScope foreachScope)
         }
     }
 
+    @Override
+    public void setMacroLibraries(List<Template> macroLibraries)
+    {
+        // Not super clean but that's the only way to not end up with current macro libraries being broken by
+        // Templace#parse
+        // TODO: remove when https://issues.apache.org/jira/browse/VELOCITY-967 is fixed
+        if (macroLibraries == null || !macroLibraries.isEmpty()) {
```

Comment:
The condition is weird but I guess it's expected.

Reply:
Yes, it's weird. Not super happy about that one, but we'll see if Velocity guys accept my suggestion.